src/meta/meta_bulk_load_service.cpp

```diff
@@ -1600,9 +1623,18 @@ void bulk_load_service::on_query_bulk_load_status(query_bulk_load_rpc rpc)
     }
 
     response.bulk_load_states.resize(partition_count);
+    response.__set_hp_bulk_load_states(
+        std::vector<std::map<host_port, partition_bulk_load_state>>(partition_count));
     for (const auto &kv : _partitions_bulk_load_state) {
         if (kv.first.get_app_id() == app_id) {
-            response.bulk_load_states[kv.first.get_partition_index()] = kv.second;
+            auto gpid = kv.first.get_partition_index();
```

Comment:
gpid is a type name, it's better to avoid using it as a variable name.
```suggestion
            auto pidx = kv.first.get_partition_index();
```